iree/compiler/Conversion/CodegenUtils/FunctionUtils.cpp

```diff
@@ -16,6 +16,17 @@ namespace iree_compiler {
 
 bool isEntryPoint(FuncOp func) { return func.isPublic(); }
 
+FailureOr<FuncOp> getSingleEntryPointFunction(ModuleOp module) {
```

Comment:
Is this a hard requirement for some reason? I'd really like to not add back the same restrictions we had on the old code path - being able to have multiple functions - especially on the CPU - is really important. It's an artifact of where we do the final-phase linking that there's only single functions at this point in the lowering, but that's not a requirement - someone could come in with a flow.executable with multiple functions. Looking below it seems like you could do the materialization in a loop over all the exported functions?

Reply:
No its not a hard requirement. We could handle multiple entry points, etc. I wrote the code first without any such requirements, but then felt like I was just conjecturing about things, without a way to verify my assumptions are correct. So bit of over-indexing on what is "known works path" vs "should work without this as well". If you feel strongly about it, I can go and revisit this and drop this.